gate-keeper/readme.md

```diff
@@ -0,0 +1,103 @@
+## Introduction
+This guide will walk you through the process of building contracts for the gate keeper, deploying them into an existing world running in Docker, and testing their functionality by executing scripts.
+
+The gate keeper allows you to crowdfund items toward a specific goal. For example, if you need a certain amount of resources to build a new ship for the corporation etc.
+
+## Deployment and Testing
+### Step 0: Deploy the gate keeper contracts to the existing world 
+First, copy the World Contract Address from the Docker logs obtained in the previous step, then run the following command:
+
+![alt text](../docker_deployment.png)
+
+```bash
+cd gate-keeper/packages/contracts
+```
+
+Install the dependecies for the contracts:
+```bash
+pnpm install
+```
+
+**Local Deployment**
+This will deploy the contracts to your local world.
+```bash
+pnpm run deploy:local --worldAddress <worldAddress> 
+```
+
+**Devnet Deployment**
+This will deploy the contracts to the Devnet world.
+```bash
+cd packages/contracts
+pnpm run deploy:devnet --worldAddress <worldAddress> 
+```
+
+
+eg: `pnpm run deploy:local --worldAddress 0xafc8e4fd5eee66590c93feebf526e1aa2e93c6c3`
+
+Once the deployment is successful, you'll see a screen similar to the one below. This process deploys the Gate Keeper contract.
+![alt text](./readme-imgs/deployment.png)
+
+
+### Step 1: Setup the environment variables 
+Next, replace the following values in the [.env](./packages/contracts/.env) file with the values you copied earlier:
+
+```bash
+#WORLD ADDRESS COPIED FROM DOCKER LOGS
+WORLD_ADDRESS=
+
+#DONT NEED TO CHANGE IF YOUR RUNNING LOCALLY
+SSU_ID=
+```
+
+You can adjust the remaining values in the .env file as needed, though they are optional.
+
+
+<details markdown="block">
+<summary>Changing optional environment values</summary>
+
+#### Setting item wanted and amount
+You can set the item you want to crowdfund and the quantity for the goal.
+
+```bash
+#Item for the goal
+INVENTORY_ITEM_ID=1135
+#Quantity needed to reach the goal
+TARGET_QUANTITY=10
+```
+
+</details>
+
+
+### Step 2: Mock data for the existing world 
+To generate mock data for testing the Gate Keeper logic, run the following command:
+
+```bash
+pnpm run mock-data
+```
+This will create the on-chain SSU, fuel it and bring it online.
```

Comment:
can we add some details to it like this SSU will act as a gatekeeper where it has a inventory and it needs to be online to deposit items into it ?
The more details the better.

Reply:
Have sorted.